pkg/operator/controller/ingress/controller.go

```diff
@@ -473,17 +473,30 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 			}
 			switch lbType {
 			case operatorv1.AWSLoadBalancerProvider:
-				// The only provider parameter that is supported
-				// for AWS is the connection idle timeout for
-				// classic ELBs.
-				var specIdleTimeout, statusIdleTimeout metav1.Duration
-				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					specIdleTimeout = specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					statusIdleTimeout = statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if specIdleTimeout != statusIdleTimeout {
+				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
+					if statusLB == nil {
+						statusLB.ProviderParameters = specLB.ProviderParameters
+						changed = true
+					} else if statusLB != nil && statusLB.ProviderParameters == nil {
+						statusLB.ProviderParameters = specLB.ProviderParameters
+						changed = true
+					} else if statusLB != nil && statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS == nil {
+						statusLB.ProviderParameters = specLB.ProviderParameters
+						changed = true
+					} else if statusLB != nil && statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil {
+						statusLB.ProviderParameters = specLB.ProviderParameters
+						changed = true
+					} else if statusLB != nil && statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
```

Comment:
When we use ingress controller with CLB the ClassicLoadBalancerParameters can be nil or not nil. Eg- The default IC with clb one gets created with  no  ClassicLoadBalancerParameters i.e nil. So having ClassicLoadBalancerParameters is not mandatory for user to set it in the spec of the IC when he wants CLB.

Reply:
When we IC has CLB then the ClassicLoadBalancerParameters is not mandatory. Eg- the default IC with CLB has no ClassicLoadBalancerParameters. So it can be set or not set.  So, we need to check for `statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters` for nil and not nil along with the value set by the user in the spec for IC which in this is NLB and then set the provider parameters of the spec to the status which then is used by the lb service code.

Reply:
Got it thanks for the details, that makes more sense. But checking for something that is nil or not nil is always going to enter one of this `else if` statements, it can't be anything else other than nil or not nil, so what's the point in checking `ClassicLoadBalancerParameters` in the first place? Aren't you effectively doing `exists or not exists` which is always true?

Reply:
To further my point, if `statusLB.ProviderParameters.AWS != nil`, what logic would make it past this point? `statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters` would always be either `nil` or not `nil`. So the `if else` statements below appear to be dead code, never able to be reached.